dsc_lib/src/configure/mod.rs

```diff
@@ -240,23 +238,29 @@ impl Configurator {
         let mut conf = config_doc::Configuration::new();
 
         for resource in &config.resources {
-            let Some(dsc_resource) = self.discovery.find_resource(&resource.resource_type).next() else {
+            let Some(dsc_resource) = self.discovery.find_resource(&resource.resource_type.to_lowercase()) else {
                 return Err(DscError::ResourceNotFound(resource.resource_type.clone()));
             };
-            add_resource_export_results_to_configuration(&dsc_resource, &mut conf)?;
+            add_resource_export_results_to_configuration(dsc_resource, &mut conf)?;
         }
 
         result.result = Some(conf);
 
         Ok(result)
     }
 
-    fn validate_config(&self) -> Result<(Configuration, Vec<ResourceMessage>, bool), DscError> {
+    fn validate_config(&mut self) -> Result<(Configuration, Vec<ResourceMessage>, bool), DscError> {
         let config: Configuration = serde_json::from_str(self.config.as_str())?;
         let mut messages: Vec<ResourceMessage> = Vec::new();
         let mut has_errors = false;
+        
+        // Perform discovery of resources used in config
+        let required_resources = config.resources.iter().map(|p| p.resource_type.to_lowercase()).collect::<Vec<String>>();
```

Comment:
This doesn't prune duplicates right?  We should ensure the list is unique before discovery

Reply:
Good point;  Added removal of duplicated.